### PR TITLE
[DUOS-2811][risk=no] Enable data submitter dataset deletion via the UI on data submissions

### PR DESCRIPTION
**Pull request by aarohinadkarni**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2811

### Summary

Adds a delete confirmation modal for the deletion of data submissions not yet accepted on the Data Submissions tab.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/researcher_console/DatasetSubmissionsTable.jsx [hunk 1]
```diff
@@ -65,11 +68,49 @@ export default function DatasetSubmissionsTable(props) {
...
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const removeDataset = async (termToDelete) => {
+    const termName = termToDelete.datasetName;
+    const termId = termToDelete.datasetId;
+    let updatedTerm = await DataSet.deleteDataset(termId).then(() => {
+        window.location.reload();
+    });
```
Comment (rushtong) [inline src/pages/researcher_console/DatasetSubmissionsTable.jsx]: This causes a refresh while the dialog is open which is slightly confusing as a user - I wasn't clear on what was happening.

Comment (rushtong) [reply src/pages/researcher_console/DatasetSubmissionsTable.jsx]: In the error case, this doesn't close the dialog - it remains open. The error case happens when a dataset is in use. Look for the `deletable` property on the dataset. On the back end, we query the DAR-Dataset join table to see if the dataset has been requested. If it is in use, we mark this field as false.

File: src/pages/researcher_console/DatasetSubmissionsTable.jsx [hunk 2]
```diff
@@ -89,6 +130,24 @@ export default function DatasetSubmissionsTable(props) {
...
+            className={'glyphicon glyphicon-trash'}
+            onClick={() => handleClick(term)}
+            to={`#`}
+          />
+          <ConfirmationDialog 
+            title="Delete dataset" 
+            openState={open} 
+            close={handleClose}
+            action={() => removeDataset(selectedTerm)}
+            description={`Are you sure you want to delete the dataset named '${selectedTerm.datasetName}'?`}
```
Comment (rushtong) [inline src/pages/researcher_console/DatasetSubmissionsTable.jsx]: Nit: I suggest using the dataset identifier here - names in production are extremely long and will affect the modal display here.

**Review by rushtong — COMMENTED** [review]
See comments inline. Let's figure out which approach we want to take and consolidate our approaches.

File: src/components/modals/ConfirmationDialog.js [hunk 3]
```diff
@@ -30,4 +30,4 @@ export const ConfirmationDialog = (props) => {
             </DialogActions>
         </Dialog>
     );
-}
+}
```
Comment (rushtong) [inline src/components/modals/ConfirmationDialog.js]: nit: revert this to minimize unnecessary changes

**Review by rushtong — APPROVED** [review]
Thanks for this, it looks good! One minor suggestion, then 👍🏽

**Review by fboulnois — APPROVED** [review]
looks good 👍
